packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,1237 @@
 import { CfnGatewayRoute } from './appmesh.generated';
+import { HttpHeaderMatch } from './route-spec';
 import { Protocol } from './shared-interfaces';
 import { IVirtualService } from './virtual-service';
 
 // keep this import separate from other imports to reduce chance for merge conflicts with v2-main
 // eslint-disable-next-line no-duplicate-imports, import/order
 import { Construct } from '@aws-cdk/core';
 
+
+/**
+ * Configuration for gRPC gateway route match with the host name.
+ */
+export interface GatewayRouteHostnameMatchConfig {
+  /**
+   * GatewayRoute CFN configuration for host name match.
+   */
+  readonly hostnameMatch: CfnGatewayRoute.GatewayRouteHostnameMatchProperty;
+}
+
+/**
+ * Used to generate host name matching methods.
+ */
+export abstract class GatewayRouteHostname {
+  /**
+   * The value of the host name must match the specified value exactly.
+   *
+   * @param name The exact host name to match on
+   */
+  public static matchingExactly(name: string): GatewayRouteHostname {
```

Comment:
This may be a better question for Adam, but why do we do `matching...` and not `match...`, i.e. `matchExact` or `matchSuffix`?

Reply:
This naming convention is same as how we implemented `SubjectAlternativeNames` factory method that was suggested by Adam..

I did not ask him for detailed reasoning but I think it really believe `matching...` or `match` would not have much different.

However, I Adam did mean to avoid calling it like `...Match` tho.